### PR TITLE
Viewport: export context for compatibility with hooks

### PR DESCRIPTION
**Pull request by AquiGorka**

### CODE REVIEW

**Comment by bpierre** [comment]
I was thinking of exporting a hook directly for this:

`const { width, height } = useViewport()`

What do you think?

**Comment by AquiGorka** [comment]
We could include those two props into the same `Viewport` context no?

**Comment by bpierre** [comment]
Yes they all are in the context, but I see the context more as an implementation thing rather than something we should have as a public API.

I think it would be nice to provide hooks equivalents for every component in `providers/`: https://github.com/aragon/aragon-ui/tree/master/src/providers

**Comment by AquiGorka** [comment]
Super, in the meantime, I added these changes so that we can start using `Viewport` with hooks (to make code easier on the 👀)

**Comment by bpierre** [comment]
But if we export the context we’ll have to support it, and exposing contexts is not the kind of API we have elsewhere in aragonUI… why not adding a hook directly?

**Comment by AquiGorka** [comment]
Ahhh, now I understand what you were saying: you don't want to expose the `context` but rather the custom hook.

**Comment by AquiGorka** [comment]
Updated: context not exported and exports a hook as suggested.

**Comment by bpierre** [comment]
Awesome! We should also export it here:

https://github.com/aragon/aragon-ui/blob/master/src/providers/index.js#L6

**Comment by AquiGorka** [comment]
@bpierre listo 😄

File: src/providers/Viewport/README.md [hunk 1]
```diff
@@ -6,6 +6,24 @@ to help building responsive apps.
...
+      {within('medium', 'large') && <div>medium</div>}
+      {above('large') && <div>large</div>}
+    </div>
+  )
+}
+```
+
+### Render prop
+
 ```jsx
```
Comment (bpierre) [inline src/providers/Viewport/README.md]: Could we add the import + wrap in in `MyComponent` here, to highlight the difference between the two?

Comment (AquiGorka) [reply src/providers/Viewport/README.md]: Sure. Done.

**Pull request merged**